### PR TITLE
MueLu: interfacing Structured_Driver to more general iterative solver

### LINKED ISSUE #5198: MueLu: adding more general interface to belos in structured test
## Enhancement
@trilinos/muelu 

Switch from using a `GmresSolverMgr` to using a `SolverFactory` which allows users to switch between the different iterative solvers in Belos instead of only using GMRES.

### PR DESCRIPTION
**Pull request by lucbv**
@trilinos/muelu 

## Description
switch from Gmres solver manager to solver factory in Structured_Driver

## Motivation and Context
This just allows a lot more flexibility in the testing that can be done using command line arguments.

## Related Issues

* Closes #5198
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 

## How Has This Been Tested?
I performed a local build and test

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

**Comment by lucbv** [comment]
Doing a small clean-up and switching the base to develop...

File: packages/muelu/test/structured/Driver_Structured.cpp [hunk 1]
```diff
@@ -152,7 +152,7 @@ int main_(Teuchos::CommandLineProcessor &clp, Xpetra::UnderlyingLib& lib, int ar
   double      tol               = 1e-12;             clp.setOption("tol",                   &tol,               "solver convergence tolerance");
   bool        scaleResidualHist = true;              clp.setOption("scale", "noscale",      &scaleResidualHist, "scaled Krylov residual history");
   bool        solvePreconditioned = true;            clp.setOption("solve-preconditioned","no-solve-preconditioned", &solvePreconditioned, "use MueLu preconditioner in solve");
-  std::string operation         = "solve";           clp.setOption("op",                    &operation,         "operation to perform: (matvec | setup | solver)");
+  std::string operation         = "solve";           clp.setOption("op",                    &operation,         "operation to perform: (matvec | setup | solver)");  std::string belosType         = "cg";              clp.setOption("belosType",             &belosType,         "belos solver type: (Pseudoblock CG | Block CG | Pseudoblock GMRES | Block GMRES | ...) see BelosSolverFactory.hpp for exhaustive list of solvers");
```
Comment (mhoemmen) [inline packages/muelu/test/structured/Driver_Structured.cpp]: Did you mean not to have any line breaks here?  Anyway, no biggie :-)

Comment (lucbv) [reply packages/muelu/test/structured/Driver_Structured.cpp]: No I want a break line and it appears in emacs (I actually was surprised and double checked...).
I am not too sure why github does not see it though?

Comment (mhoemmen) [reply packages/muelu/test/structured/Driver_Structured.cpp]: Huh, weird.

**Pull request merged**